pudb/debugger.py

```diff
@@ -2799,11 +2804,12 @@ def make_frame_ui(frame_lineno):
                     ui_log.exception(message)
                     class_name = "!! %s !!" % message
 
-            return StackFrame(frame is self.debugger.curframe,
+            return StackFrame(i == self.debugger.curindex,
                     code.co_name, class_name,
                     self._format_fname(code.co_filename), lineno)
 
-        frame_uis = [make_frame_ui(fl) for fl in self.debugger.stack]
+        frame_uis = [make_frame_ui(i, fl)
+                for i, fl in enumerate(self.debugger.stack)]
```

Comment:
Nice! I recall encountering duplicated selection in the stack view in other contexts as well, this should fix all of that up!